lib/mysql.go

```diff
@@ -384,22 +412,25 @@ func (m *MySQLPlugin) FetchMetrics() (map[string]float64, error) {
 	if m.isUnixSocket {
 		proto = "unix"
 	}
-	db := mysql.New(proto, "", m.Target, m.Username, m.Password, "")
-	switch c := db.(type) {
-	case *native.Conn:
-		c.Debug = m.Debug
-	}
-	err := db.Connect()
+
+	// 	username:password@unix(/var/lib/mysql/mysql.sock)/dbname?param=value
+	dsn := fmt.Sprintf("%s:%s@%s(%s)/", m.Username, m.Password, proto, m.Target)
```

Comment:
There is a way to generate dns strings using `Config` from github.com/go-sql-driver/mysql package. It is not required, but may improve the outlook of the code.

```go
cnf := mysql.NewConfig()
cnf.User = m.Username
cnf.Passwd = m.Password
cnf.Net = proto
cnf.Addr = m.Target
// e.g.) "username:password@unix(var/lib/mysql/mysql.sock)"
dsn := cnf.FormatDSN()
```